agreement/proposalTracker_test.go

```diff
@@ -63,36 +63,50 @@ func TestProposalTrackerProposalSeeker(t *testing.T) {
 	var err error
 	assert.False(t, s.Frozen)
 	assert.False(t, s.Filled)
+	assert.False(t, s.hasLowestAfterFreeze)
 
 	// issue events in the following order: 2, 3, 1, (freeze), 0
-	s, err = s.accept(votes[2])
+	var effect CredentialTrackingEffect
+	s, effect, err = s.accept(votes[2])
 	assert.NoError(t, err)
+	assert.Equal(t, effect, VerifiedBetterCredentialForTracking)
 	assert.False(t, s.Frozen)
 	assert.True(t, s.Filled)
 	assert.True(t, s.Lowest.equals(votes[2]))
+	assert.True(t, s.hasLowestAfterFreeze)
 
-	s, err = s.accept(votes[3])
+	s, effect, err = s.accept(votes[3])
 	assert.Error(t, err)
+	assert.Equal(t, effect, NoCredentialTrackingImpact)
 	assert.False(t, s.Frozen)
 	assert.True(t, s.Filled)
 	assert.True(t, s.Lowest.equals(votes[2]))
+	assert.True(t, s.hasLowestAfterFreeze)
 
-	s, err = s.accept(votes[1])
+	s, effect, err = s.accept(votes[1])
 	assert.NoError(t, err)
+	assert.Equal(t, effect, VerifiedBetterCredentialForTracking)
 	assert.False(t, s.Frozen)
 	assert.True(t, s.Filled)
 	assert.True(t, s.Lowest.equals(votes[1]))
+	assert.True(t, s.hasLowestAfterFreeze)
 
+	lowestBeforeFreeze := s.Lowest
 	s = s.freeze()
 	assert.True(t, s.Frozen)
 	assert.True(t, s.Filled)
 	assert.True(t, s.Lowest.equals(votes[1]))
```

Comment:
minor: I think it'd be a good idea to assert that `s.lowestAfterFreeze` equals `s.Lowest` for the pre-freezing cases. Saying this because `player.updateCredentialArrivalHistory` _only_ looks at `LowestAfterFreeze` and assumes this value will be kept up-to-date before freezing as well.